### PR TITLE
docs: Restructure ruby docs

### PR DESCRIPTION
**Pull request by svrnm**
Hi there,

this PR introduces two changes into the `website_docs` (let me know if there should be 2 PRs):

* Move pages on events & context propagation into the page about manual instrumentation, this is then similar to how other languages have it
* Use the new shortcode `lang_instrumentation_index_head` we introduced recently for opentelemetry.io, which allows us to pull in status values from a single source, so we can make it reusable

### CODE REVIEW

**Review by arielvalentin — APPROVED** [review]

**Comment by svrnm** [comment]
> @svrnm We use conventional commits https://www.conventionalcommits.org/en/v1.0.0/
> 
> Please amend the commit with `docs:` prefix.

of course! Fixed that.

File: website_docs/manual.md
Comment (plantfansam) [inline website_docs/manual.md:4]: `manual_instrumentatio` -> `manual_instrumentation` ?

**Review by plantfansam — COMMENTED** [review]
one nit, but lgtm

File: website_docs/manual.md
Comment (svrnm) [reply website_docs/manual.md:4]: of course, thanks!

Comment (svrnm) [reply website_docs/manual.md:4]:
```suggestion
aliases: [/docs/instrumentation/ruby/manual_instrumentation, /docs/instrumentation/ruby/events, /docs/instrumentation/ruby/context-propagation]
```

**Comment by svrnm** [comment]
cc: @chalin , @cartermp

File: website_docs/_index.md
Comment (chalin) [inline website_docs/_index.md:21]: Let's drop the comment as we agreed to do for https://github.com/open-telemetry/opentelemetry.io/pull/1861.

File: website_docs/manual.md
Comment (chalin) [inline website_docs/manual.md:4]: Once the list becomes too long, it's probably best to change the sytnax:

```suggestion
aliases:
  - /docs/instrumentation/ruby/manual_instrumentation
  - /docs/instrumentation/ruby/events
  - /docs/instrumentation/ruby/context-propagation
```

**Review by chalin — APPROVED** [review]
See inline comments.

(@svrnm - have you tested these changes through the website to ensure that it builds successfully? Not a requirement, just curious.)

**Review by cartermp — APPROVED** [review]

File: website_docs/_index.md
Comment (svrnm) [reply website_docs/_index.md]: removed

**Review by plantfansam — APPROVED** [review]

**Comment by svrnm** [comment]
@open-telemetry/ruby-approvers  anything that blocks this from merging? TY.

**Review by robertlaurin — APPROVED** [review]

**Pull request merged**
